### PR TITLE
feat(sdk): Add namesapce grants to key splitting

### PR DESCRIPTION
**Pull request by elizabethhealy**
resolves https://github.com/opentdf/platform/issues/1516

### CODE REVIEW

File: sdk/granter_test.go [hunk 1]
```diff
@@ -552,3 +596,115 @@ func TestReasonerSpecificity(t *testing.T) {
 		})
 	}
 }
+
+func TestReasonerSpecificityWithNamespaces(t *testing.T) {
```
Comment (strantalis) [inline sdk/granter_test.go:600]: Can we update the table test message to make it clear what is happening and what the expected output should be.

**Comment by elizabethhealy** [comment]
successful xtest run https://github.com/opentdf/tests/actions/runs/10724372814/job/29739869061?pr=189

**Review by dmihalcik-virtru — APPROVED** [review]

File: sdk/granter_test.go (hunk 1 above)
Comment (strantalis) [reply sdk/granter_test.go:611]: @elizabethhealy I should have provided an example but I was thinking like `expect namespace kas when no value or definition grant`

**Review by strantalis — CHANGES_REQUESTED** [review]

File: sdk/granter_test.go (hunk 1 above)
Comment (elizabethhealy) [reply sdk/granter_test.go:626]: updated

**Review by strantalis — APPROVED** [review]

**Pull request merged**